src/darwin/frida-helper-backend-glue.m

```diff
@@ -1325,19 +1326,44 @@ static void frida_darwin_helper_backend_launch_using_sbs (NSString * identifier,
 frida_darwin_helper_backend_is_application_process (guint pid)
 {
   gboolean result = FALSE;
+  gchar path[4 * MAXPATHLEN];
+  gchar * dirname;
+  struct stat info;
   NSAutoreleasePool * pool;
-  NSString * identifier;
+  NSURL * plist_url;
+  NSDictionary * plist = nil;
+  NSString * identifier = nil;
+
+  if (proc_pidpath (pid, path, sizeof (path)) <= 0)
+    return result;
+
+  dirname = g_path_get_dirname (path);
+  g_strlcpy (path, dirname, sizeof (path));
+  g_free (dirname);
+
+  g_strlcat (path, "/Info.plist", sizeof (path));
```

Comment:
Perhaps it would be simpler to chain `URLByDeletingLastPathComponent` and `URLByAppendingPathComponent:isDirectory:` and let the `plist == nil` error-handling take care of the rest.